src/main/java/com/fcgl/madrid/points/service/UserActionService.java

```diff
@@ -1,22 +1,70 @@
 package com.fcgl.madrid.points.service;
 
+import com.fcgl.madrid.points.model.Action;
+import com.fcgl.madrid.points.model.Trophy;
 import com.fcgl.madrid.points.model.UserAction;
+import com.fcgl.madrid.points.model.UserTrophy;
+import com.fcgl.madrid.points.repository.ActionRepository;
+import com.fcgl.madrid.points.repository.TrophyRepository;
 import com.fcgl.madrid.points.repository.UserActionRepository;
+import com.fcgl.madrid.points.repository.UserTrophyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
 public class UserActionService {
     private UserActionRepository userActionRepository;
+    private ActionRepository actionRepository; // Is this the correct way to access the action table?
+    private TrophyRepository trophyRepository;
+    private UserTrophyRepository userTrophyRepository;
 
     @Autowired
-    public UserActionService(UserActionRepository userActionRepository) {
+    public UserActionService(UserActionRepository userActionRepository,
+                             ActionRepository actionRepository,
+                             TrophyRepository trophyRepository,
+                             UserTrophyRepository userTrophyRepository) {
         this.userActionRepository = userActionRepository;
+        this.actionRepository = actionRepository;
+        this.trophyRepository = trophyRepository;
+        this.userTrophyRepository = userTrophyRepository;
     }
 
     public List<UserAction> findAll() {
         return userActionRepository.findAll();
     }
+
+    public ResponseEntity<Trophy> addUserAction(Long userID, Long actionID) {
+
+        // Check that the action exists
+        Action action = actionRepository.findById(actionID).orElse(null);
+        if (action == null) {
+            return ResponseEntity.badRequest().body(null);
+        }
+
+        UserAction userAction = userActionRepository.findByUserIdAndAction(userID, action).orElse(null);
+        if (userAction == null) {
+            // We haven't seen this action for this user,
+            // Let's create a new entry with count equal to 1
+            userAction = new UserAction(userID, action, 1);
+            userActionRepository.save(userAction);
+        } else {
+            // We have seen this action for this user before
+            // just increment the action counter
+            userAction.setCount(userAction.getCount() + 1);
+            userActionRepository.save(userAction);
+        }
+
+        // TODO: Update user points...
+
+        Trophy trophy = trophyRepository.findByAction(action).orElse(null);
+        if (trophy != null) {
+            UserTrophy userTrophy = new UserTrophy(userID, trophy);
```

Comment:
Need to validate that the UserAction `count` is equal to the Trophy `numberOfPoints` before adding a UserTrophy

Reply:
done. I also added trophy count to account for multiple earned trophies.

Check https://github.com/fcgl/point-system/pull/9/commits/29c66a169d71ffd9447a6007ae1a73d864f80fd0